tests/common.go

```diff
@@ -8198,6 +8209,86 @@ func StartTorpedoTest(testName, testDescription string, tags map[string]string,
 		RunIdForSuite = testrailuttils.AddRunsToMilestone(testRepoID)
 		CurrentTestRailTestCaseId = testRepoID
 	}
+	log.Infof("TOGGLE_PURE_MGMT_IP: %v", os.Getenv("TOGGLE_PURE_MGMT_IP"))
+	if os.Getenv("TOGGLE_PURE_MGMT_IP") != "" {
+		if PureMgmtIpCounter == 0 {
+			// Fetch the pure.json file and  implement methods to store FA multiple management endpoints in a map
+			// PureMgmtIPList consists List of Pure FA MGMT endpoints
+			// PureFAMgmtMap is a map and consists of FA MGMT IP as key and its corresponding client as value
+			// PureMgmtIpCounter is tracker which increments after each test case
+			volDriverNamespace, err := Inst().V.GetVolumeDriverNamespace()
+			log.FailOnError(err, "failed to get volume driver [%s] namespace", Inst().V.String())
+			secret, err = pureutils.GetPXPureSecret(volDriverNamespace)
+			log.FailOnError(err, "failed to get secret [%s/%s]", PureSecretName, volDriverNamespace)
+			PureFAMgmtMap, err = pureutils.GetFAMgmtIPFromPXPureSecret(secret)
+			log.FailOnError(err, "failed to get FA management map from secret [%s/%s]", PureSecretName, volDriverNamespace)
+			for mgmtIP := range PureFAMgmtMap {
+				PureMgmtIPList = append(PureMgmtIPList, mgmtIP)
+			}
+			faMgmtIP := PureMgmtIPList[PureMgmtIpCounter]
+			faClient := PureFAMgmtMap[faMgmtIP]
+			apiToken := pureutils.GetApiTokenForMgmtEndpoints(secret, faMgmtIP)
+			// Loop through Network interfaces to find the VIF interface which we use in API calls for enabling/disabling interfaces
```

Comment:
This part especially is a good candidate to extract to a function: "LookupVIF" or something